### DISCUSSION TITLE
A few considerations on 4.6.0(alpha+beta+rc01), alpha releases and ARM64, Debian and this repo's Readme.md s

### DISCUSSION THREAD

**Discussion by fabiorzfreitas** [qbittorrent/docker-qbittorrent-nox#12] [General]
Yeah, I know, I mentioned way to many topics and while for my own usage I'll need to figure out the rest, discussing these topics may be useful for the community, so let's get into it:

- [This single commit](https://github.com/qbittorrent/qBittorrent/commit/e37661d53a0249548238a89d41736260998348f6) for 4.6.0 alpha, beta and rc01 is the only thing I'm looking for. I want that improved logging on the UI, I've had a really bad time figuring out the errors I was getting (while using `linuxserver/qbittorrent`'s image) without proper logs on an "errored" torrent;

- This means I became interested in finding other images that perhaps could allow me to pull that release, that's how I got here. I was halted by a smaller rock than the one that would come after, but initially, I was very unhappy to find out that the `:alpha` tag only had a release for amd64, but I needed an ARM64 version. That's when I began inspecting the code here and in a few other repos, plus some parts of Docker's documentation, because, on top of everything, my knowledge on Docker is somewhat limited, I let Portainer do the rough work;

- I couldn't figure out what was the reasoning of this restriction on the alpha tag. I saw the (bot maintained) alpha.yaml, but could not really figure it out, so here's an actual question: Why alpha releases only support amd64? Could I change it if I were to fork this repo, or would I simply have spent a lot of time and effort on something I could have asked about first?

- Now comes the big rock: reading the Dockerfile's contents immediately raised a flag, `Alpine` could potentially be an issue. I was checking for compatibility, as my hardware is a Raspberry Pi 4, 8GB with OMV on the top of Raspberry Pi OS. My first thought was to check the distro tree, maybe Alpine and Debian could be friends. They could not and I think I confirmed it by, for the first time, trying to deploy this package's latest image (I actually modified the compose.yaml a bit so I wouldn't have to deal with a .env file. I got some errors on this, but managed to proceed with the deployment). Well, it didn't work (I just assumed everything would work fine like linuxserver's does). Perhaps I f'ck up with my changes on compose.yaml, but the errors thrown at me became wildly different than my previous messy vars.

- Luckily for me, I came across with this awesome repo, [userdocs/qbittorrent-nox-static](https://github.com/userdocs/qbittorrent-nox-static), which I figured could solve (or at least help me by studying the code) that big problem. Yet, their code relies on this repo's `:latest`, so once again I had to check out about amd64 vs ARM64 before I started playing Dr. Frankenstein!

- It seems I only have one *real* question, but I'd also like to use this moment to point that the `Readme.md` file on `manual_build` was almost identical to the main one, I was actually expecting to find instructions for a custom Dockerfile or something like that. I'm not sure about what should be done here, if the contents from both should be split or if `manual_build` should be removed at all, but I found this little thing and figured it wasn't enough for an issue, but I was on the verge of starting this discussion, so here it is!

**tl-dr;** I need one single commit from any of the 4.6.0's pre-releases and that would require me to use `:alpha`, but it seems it's only for amd64, while I'm using ARM64. Is it something I can fork and try to solve myself or would it be a fool's errand?

Thank you :)

**Comment by Chocobo1** [comment]
>tl-dr; I need one single commit from any of the 4.6.0's pre-releases and that would require me to use :alpha, but it seems it's only for amd64, while I'm using ARM64. Is it something I can fork and try to solve myself or would it be a fool's errand?

You could try to replace this line 
https://github.com/qbittorrent/docker-qbittorrent-nox/blob/d75e597f431525762a4607ac4ca4d06acc76817c/.github/workflows/alpha.yaml#L19
with
`- "linux/arm64/v8"`
and run the CI script.

>Why alpha releases only support amd64?

Based on the download statistics I've seen, there aren't many users outside of amd64 and building ARM variations are quite slow.

**Reply by fabiorzfreitas** [reply]
Great news, thank you! I would have to replace this line on both `alpha.yaml` and `CI.yaml`, right?
